src/pages/index.js

```diff
@@ -1,62 +1,169 @@
+    userID = _id;
+    userInfo.setUserInfo(name, about);
+    userInfo.setUserProfileImage(avatar);
+  })
+  .catch((error) => {
+    console.error("Error fetching data:", error);
+  });
 
 // functions
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
+  const { name, about } = data;
+  api
+    .updateProfile(name, about)
+    .then(() => {
+      profileEditPopup.editSubmitBtn();
+    })
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+      profileEditPopup.resetSubmitBtn();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
 }
 
 function handleCardFormSubmit(data) {
-  const name = data.title;
-  const link = data.imageUrl;
-  const card = createCard({ name, link });
-  cardSection.prependItem(card);
-  profileAddPopup.close();
+  api.postCard(data).then((data) => {
+    const card = createCard(data);
+    cardSection.prependItem(card);
+    profileAddPopup.close();
+    profileAddPopup.resetSubmitBtn();
+  });
+}
+
+function handleDeleteIcon(cardId, card) {
+  deleteCardPopup.open(cardId);
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        deleteCardPopup.close();
+        card.remove();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleProfileImgFormSubmit(data) {
+  api
+    .updateProfileImg(data.link)
+    .then(() => {
+      userInfo.setUserProfileImage(data.link);
+      profileImagePopup.close();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
 }
 
 function handleCardClick(name, link) {
   previewImagePopup.open(name, link);
 }
+function addLike(cardId, likeCounter) {
+  api
+    .addLike(cardId)
+    .then((res) => {
+      likeCounter.textContent = res.likes.length;
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
+}
+
+function deleteLike(cardId, likeCounter) {
+  api
+    .deleteLike(cardId)
+    .then((res) => {
+      likeCounter.textContent = res.likes.length;
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
+}
+
+function handleCardLike(cardId, isLiked, likeCounter) {
+  if (isLiked) {
+    deleteLike(cardId, likeCounter);
+  } else {
+    addLike(cardId, likeCounter);
+  }
+}
 
 function createCard(cardData) {
-  const newCard = new Card(cardData, "#card-template", handleCardClick);
+  const newCard = new Card(
+    cardData,
+    "#card-template",
+    userID,
+    handleCardClick,
+    handleCardLike,
+    handleDeleteIcon
+  );
   return newCard.getView();
 }
 
-// section
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const card = createCard(item);
-      cardSection.appendItem(card);
+// render initial cards
+let cardSection;
+api.getInitialCards().then((result) => {
```

Comment:
* You need to combine the requests of the user's info and the cards together in `Promise.all`